src/lib/contracts.ts

```diff
@@ -51,13 +94,30 @@ function isValidRequirementType(
 	return requirement in requirementVersions;
 }
 
+function partition<T>(arr: T[], predicate: (t: T) => boolean) {
+	return arr.reduce(
+		({ pass, fail }, elem) => {
+			if (predicate(elem)) {
+				pass.push(elem);
+			} else {
+				fail.push(elem);
+			}
+			return { pass, fail };
+		},
+		{ pass: [] as T[], fail: [] as T[] },
+	);
+}
+
 export function containerContractsFulfilled(
-	serviceContracts: ServiceContracts,
-): ApplicationContractResult {
-	const containers = _(serviceContracts).map('contract').compact().value();
+	servicesWithContract: ServiceWithContract[],
+): AppContractResult {
+	const containers = servicesWithContract
+		.map(({ contract }) => contract)
+		.filter((c) => !!c) as ContractObject[];
```

Comment:
Can you make this more explicit as either
```suggestion
		.filter((c) => c != null) as ContractObject[];
```
or
```suggestion
		.filter((c) => c === true) as ContractObject[];
```
or whatever the true intent is. If you do that I think you'd also be able to drop the `as ContractObject[]` since latest typescript can infer the type guard typing that I mentioned above. I expect the true final form to be:
```suggestion
		.filter((c) => c != null);
```